data/data/gcp/master/main.tf

```diff
@@ -25,3 +25,19 @@ resource "google_compute_instance" "master" {
 
   labels = var.labels
 }
+
+# Not ideal, machine API would need to keep membership up to date
+resource "google_compute_instance_group" "master" {
```

Comment:
there should be an instance group for each Zone, and putting instances for that zone in the corresponding instance.

Reply:
That is how I started. My concern is that the number of masters must be <= the number of zones in the region with the code I have below. It works for master_count=1 and for master_count=3 as long as there are 3 zones in the region. Take a look at the latest and lmk.

Reply:
Fixed... one group per zone, and instance membership is based on instance zone.